src/core/AutomationPattern.cpp

```diff
@@ -417,33 +484,33 @@ float *AutomationPattern::valuesAfter( const MidiTime & _time ) const
 
 
 
-void AutomationPattern::flipY( int min, int max )
+void AutomationPattern::flipY(int min, int max)
 {
-	timeMap tempMap = m_timeMap;
-	timeMap::ConstIterator iterate = m_timeMap.lowerBound(0);
-	float tempValue = 0;
+	QMutexLocker m(&m_patternMutex);
 
-	int numPoints = 0;
+	timeMap::const_iterator it = m_timeMap.lowerBound(0);
 
-	for( int i = 0; ( iterate + i + 1 ) != m_timeMap.end() && ( iterate + i ) != m_timeMap.end() ; i++)
-	{
-		numPoints++;
-	}
+	if (it == m_timeMap.end()) return;
 
-	for( int i = 0; i <= numPoints; i++ )
-	{
+	float tempValue = 0;
+	float outValueOffset = 0;
 
-		if ( min < 0 )
+	do
+	{
+		if (min < 0)
 		{
-			tempValue = valueAt( ( iterate + i ).key() ) * -1;
-			putValue( MidiTime( (iterate + i).key() ) , tempValue, false);
+			tempValue = valueAt(it.key()) * -1;
```

Comment:
I think it'd be nice to overload operator*(AutomationNode, int) for AutomationNode instead of having to do this on separate lines.

Reply:
I overloaded 4 of the compound assignment operators (+=, -=, *= and /=) for consistency, even though only one is used. I made it so they work with float values instead of int, just so they can be used for other purposes besides flipping the values in the future.

Reply:
+1 on doing it for float instead of int and implementing all 4. This should make rescaling a pattern really easy,, maybe use it there and fix the fact that out points are lost?